### PR TITLE
MAGE-349 - Problem with unserialize

### PR DESCRIPTION
**Pull request by FatchipRobert**

### CODE REVIEW

File: app/code/community/Payone/Core/Model/Domain/Protocol/TransactionStatus.php [hunk 1]
```diff
@@ -331,7 +331,13 @@ public function setStatusError()
     public function getRawRequestArray()
     {
         if(!empty($this->_data['raw_request'])) {
-            $aRaw = unserialize($this->_data['raw_request']);
+            $aRaw = json_decode($this->_data['raw_request'], true);
+            if (!$aRaw) { // request used to be serialized in prior versions -> try to unserialize
+                $aRaw = unserialize($this->_data['raw_request']);
```
Comment (phoenix-schwarz) [inline app/code/community/Payone/Core/Model/Domain/Protocol/TransactionStatus.php]: to avoid a Notice, you should use `@unserialize` and `@json_decode`

Comment (phoenix-schwarz) [reply app/code/community/Payone/Core/Model/Domain/Protocol/TransactionStatus.php]: `    public function getRawRequestArray()
    {
        if(!empty($this->_data['raw_request'])) {
            $aRaw = @json_decode($this->_data['raw_request'], true);
            if (!$aRaw) { // request used to be serialized in prior versions -> try to unserialize
                $aRaw = @unserialize($this->_data['raw_request']);
                if (!$aRaw) { // if unserialize didnt work, try it again with utf8_decode
                    $aRaw = @unserialize(@utf8_decode($this->_data['raw_request']));
                }
            }
            if($aRaw) {
                return $aRaw;
            }
        }

        return false;
    }`

Comment (phoenix-schwarz) [reply app/code/community/Payone/Core/Model/Domain/Protocol/TransactionStatus.php]: [code.txt](https://github.com/PAYONE-GmbH/magento-1/files/1841756/code.txt)

Comment (FatchipRobert) [reply app/code/community/Payone/Core/Model/Domain/Protocol/TransactionStatus.php]: We dont want to suppress notices.
There shouldn't be any notices to begin with, but if there is a problem its important to see them.

Also json_decode doesn't write notices when the string couldnt be decoded.
But you can always make those edits to the files in your project.

Comment (phoenix-schwarz) [reply app/code/community/Payone/Core/Model/Domain/Protocol/TransactionStatus.php]: you're right - looks good for me

**Pull request merged**